src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -0,0 +1,259 @@
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Headers;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.COMPONENT;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.JAEGER;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.KAFKA_SERVICE;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_PROPERTY_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_TRACES_EXPORTER_ENV_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY;
+
+/**
+ * OpenTelemetry implementation of Tracing.
+ */
+class OpenTelemetryHandle implements TracingHandle {
+
+    static void setCommonAttributes(SpanBuilder builder, RoutingContext routingContext) {
+        builder.setAttribute("component", COMPONENT);
+        builder.setAttribute(SemanticAttributes.PEER_SERVICE, KAFKA_SERVICE);
+        builder.setAttribute(SemanticAttributes.HTTP_METHOD, routingContext.request().method().name());
+        builder.setAttribute(SemanticAttributes.HTTP_URL, routingContext.request().uri());
+    }
+
+    @Override
+    public String envName() {
+        return OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+    }
+
+    @Override
+    public String serviceName(BridgeConfig config) {
+        String serviceName = System.getenv(envName());
+        if (serviceName == null) {
+            // legacy purpose, use previous JAEGER_SERVICE_NAME as OTEL_SERVICE_NAME (if not explicitly set)
+            serviceName = System.getenv(Configuration.JAEGER_SERVICE_NAME);
+            if (serviceName == null) {
+                serviceName = config.getTracingServiceName();
+            }
+            if (serviceName != null) {
+                System.setProperty(OPENTELEMETRY_SERVICE_NAME_PROPERTY_KEY, serviceName);
+            }
+        }
+        if (serviceName != null) {
+            if (System.getenv(OPENTELEMETRY_TRACES_EXPORTER_ENV_KEY) == null && System.getProperty(OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY) == null) {
+                System.setProperty(OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY, JAEGER); // it wasn't set in script
+            }
+        }
+        return serviceName;
+    }
+
+    @Override
+    public void initialize() {
+        AutoConfiguredOpenTelemetrySdk.initialize();
+    }
+
+    private static Tracer get() {
+        return GlobalOpenTelemetry.getTracer(COMPONENT);
+    }
+
+    private SpanBuilder getSpanBuilder(RoutingContext routingContext, String operationName) {
+        Tracer tracer = get();
```

Comment:
Can't we get the `Tracer` instance just one time when we create the handle instead of each time we need a span? It's just for better readiness because AFAICS in the OpenTelemetry code, the instance is anyway created one time and then just returned.